### PR TITLE
[FIX] l10n_es_pos: Prevent journal duplicate error

### PR DESCRIPTION
**Pull request by LouisGobert**
When the l10n_es_pos module is installed, the post_init_hook attempts to create an account journal for each Spanish company using a chart template. However, the method does not verify whether the journal already exists for the company, leading to potential duplication errors.

Steps to Reproduce:
- Create a journal for a Spanish company with the code 'SINV'.
- Install the Point of Sale module and the l10n_es_pos module.
- An error is raised because the journal 'SINV' already exists.

Solution:
This fix adds a check to ensure the journal does not already exist before attempting to create it.

opw-4108466

### CODE REVIEW

**Comment by aboo-odoo** [comment]
LGTM @clbr-odoo

**Review by aboo-odoo — COMMENTED** [review]
LGTM @clbr-odoo

**Review by clbr-odoo — CHANGES_REQUESTED** [review]
Hello, 
I'd like to avoid duplication of code as much as possible.
Can you check if we can do something like this please ?
```python
def _l10n_es_pos_post_init_hook(env):
    es_companies = env.companies.filtered(lambda c: c.chart_template and c.chart_template.startswith('es_'))
    for company in es_companies:
        pos_configs = [...]
        pos_configs.setup_defaults(company)
```
Looks like to me that [setup_defaults](https://github.com/odoo/odoo/commit/e9098b47734d289de43ca06bb905042e4643b89d#diff-47fe2cb88ec081675db5d687d536c1377ec0e0d6e6fd18a163ece56621ed4178R49) won't have side effects here...
In any case the search is wrong: you can't use `('company_id', '=', company.id)` in its domain, use `*self.env['account.journal']._check_company_domain(company),`.

**Comment by LouisGobert** [comment]
Hello, good catch! I wasn't familiar with that code.

File: addons/l10n_es_pos/__init__.py [hunk 1]
```diff
@@ -6,16 +6,5 @@
...
-        simplified_inv_journal = env['account.journal'].create({
-            'type': 'sale',
-            'name': _('Simplified Invoices'),
-            'code': 'SINV',
-            'default_account_id': income_account.id if income_account else False,
-            'company_id': company.id,
-            'sequence': 30,
-        })
-        pos_configs.l10n_es_simplified_invoice_journal_id = simplified_inv_journal
+        pos_configs = env['pos.config'].search([('company_id', '=', company.id)])
```
Comment (clbr-odoo) [inline addons/l10n_es_pos/__init__.py:9]: Afaik this was correct, why did you remove it ?
```suggestion
        pos_configs = env['pos.config'].search([
            *env['pos.config']._check_company_domain(company),
        ])
```

**Review by clbr-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_es_pos/__init__.py (hunk 1 above)
Comment (LouisGobert) [reply addons/l10n_es_pos/__init__.py]: Sorry, I misunderstood the meaning of your last comment.

**Review by clbr-odoo — APPROVED** [review]
Thanks ! Please take care during the FW and check that the method still exists in each subversion. 
The code of the pos changed a lot, in particular in 17.3 if I remember correctly :slightly_smiling_face: 
@robodoo r+

**Comment by Avdhutkolhe** [comment]
Sorry 😔
